tools/c7n_gcp/c7n_gcp/resources/kms.py

```diff
@@ -187,3 +189,53 @@ def get(client, resource_info):
         @classmethod
         def _get_location(cls, resource):
             return resource["name"].split('/')[3]
+
+        @classmethod
+        def _get_location(cls, resource):
+            return resource["name"].split('/')[3]
+
+
+@resources.register('kms-location')
+class KmsLocation(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'cloudkms'
+        version = 'v1'
+        component = 'projects.locations'
+        enum_spec = ('list', 'locations[]', None)
+        scope = None
+        name = id = 'name'
+        default_report_fields = [
+            "name", "createTime"]
+        asset_type = "cloudkms.googleapis.com/Location"
+
+    def _fetch_resources(self, query):
+        super_fetch_resources = QueryResourceManager._fetch_resources
+        session = local_session(self.session_factory)
+        project = session.get_default_project()
+        return super_fetch_resources(self, {'name': 'projects/{}'.format(project)})
+
+
+@KmsLocation.filter_registry.register('keyring')
+class KmsLocationKmsKeyringFilter(Filter):
+
+    schema = type_schema('keyring', **{
+        'exist': {'type': 'boolean'}},)
+    permissions = ('cloudkms.keyRings.list',)
+
+    def process(self, resources, event=None):
+        session = local_session(self.manager.session_factory)
+        client = session.client(
+            service_name='cloudkms', version='v1', component='projects.locations.keyRings')
+        accepted_resources = []
+        expecting_exist = self.data['exist']
+
+        for resource in resources:
+            kms_key_rings = client.execute_query('list', {'parent': resource['name']})
+            kms_key_rings_exist = (kms_key_rings.get('keyRings') is not None and
```

Comment:
This logic could be simplified `bool(kms_key_rings.get(‘keyRings’)`

Reply:
fixed, thanks